frontend/public/components/overview.jsx

```diff
@@ -0,0 +1,353 @@
+  }
+
+  filterItems(items) {
+    const {filterValue} = this.state;
+
+    if (!filterValue) {
+      return items;
+    }
+
+    const filterString = filterValue.toLowerCase();
+    return _.filter(items, item => {
+      return fuzzy(filterString, _.get(item, 'metadata.name', ''));
+    });
+  }
+
+  groupByLabel(items, label) {
+    const groups = _.groupBy(items, (item) => _.get(item, ['metadata', 'labels', label], 'other'));
+    return _.map(groups, (group, name) => {
+      return {
+        name,
+        items: group
+      };
+    });
+  }
+
+  getLabelGroupOptions(items) {
+    const {groupOptions} = this.state;
+    const labelKeys = _.flatten(_.map(items, item => _.keys(item.metadata.labels)));
+    return _.reduce(labelKeys, (accumulator, key) => {
+      if (_.has(key, accumulator)) {
+        return accumulator;
+      }
+
+      return {
+        ...accumulator,
+        [key]: key
+      };
+    }, groupOptions);
+  }
+
+  buildGraphForReplicators(replicators, kind) {
+    const {pods} = this.props;
+    return _.map(replicators, replicator => {
+      const {uid: replicatorUid} = replicator.metadata;
+      const ownedPods = getOwnedResources(pods.data, replicatorUid);
+      return {
+        ...replicator,
+        kind,
+        pods: ownedPods
+      };
+    });
+  }
+
+  buildGraphForRootResources(rootResources, kind) {
+    const {replicationControllers, replicaSets} = this.props;
+
+    return _.map(rootResources, rootResource => {
+      const {uid: rootResourceUid} = rootResource.metadata;
+      // Determine the replication controllers associated with this DC
+      const ownedReplicationControllers = this.buildGraphForReplicators(getOwnedResources(replicationControllers.data, rootResourceUid), 'ReplicationController');
+      const ownedReplicaSets = this.buildGraphForReplicators(getOwnedResources(replicaSets.data, rootResourceUid), 'ReplicaSet');
+      const orderedReplicationControllers = this.sortByDeploymentVersion(ownedReplicationControllers, true);
+      const currentController = _.head(orderedReplicationControllers) || _.head(ownedReplicaSets);
+      const prevController = orderedReplicationControllers[1];
+
+      return {
+        ...rootResource,
+        currentController,
+        kind,
+        prevController,
+        replicaSets: ownedReplicaSets,
+        replicationControllers: orderedReplicationControllers,
+      };
+    });
+  }
+
+  createOverviewData() {
+    const {deploymentConfigs, deployments, loaded, statefulSets} = this.props;
+    const {selectedGroupLabel} = this.state;
+
+    if (!loaded) {
+      return;
+    }
+
+    const deploymentConfigItems = this.buildGraphForRootResources(deploymentConfigs.data, 'DeploymentConfig');
+    const deploymentItems = this.buildGraphForRootResources(deployments.data, 'Deployment');
+    const statefulSetItems = this.buildGraphForReplicators(statefulSets.data, 'StatefulSet', true);
```

Comment:
Looks like the method only takes two args?

Reply:
Oops, an artifact left from a previous implementation